demos/ASP.Net Core API Gateway/Divergent.Sales.ViewModelComposition/OrdersListViewModelAppender.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Divergent.Sales.ViewModelComposition.Events;
+using ITOps.ViewModelComposition;
+using ITOps.ViewModelComposition.Json;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Divergent.Sales.ViewModelComposition
+{
+    public class OrdersListViewModelAppender : IViewModelAppender
+    {
+        public bool Matches(RouteData routeData, string verb)
+        {
+            /*
+             * matching is a bit weak in this sample, it's designed 
+             * this way to satisfy both the Gateway and the Mvc sample
+             */
+            var controller = (string)routeData.Values["controller"];
+
+            return HttpMethods.IsGet(verb)
+                && controller.ToLowerInvariant() == "orders"
+                && !routeData.Values.ContainsKey("id");
+        }
+
+        public async Task Append(dynamic vm, RouteData routeData, IQueryCollection query)
```

Comment:
Should we pass vm here or change the approach so that the appender returns dynamic? This would prevent any cross-appender type of communication and name clashes between appenders.

In addition the structure of the end response could be controlled by the caller.